src/upgrade.cpp

```diff
@@ -0,0 +1,41 @@
+
+#include <upgrade.h>
+#include <main.h>
+#include <vector>
+
+std::string getUpgradeStatus(int64_t current, int64_t changeHeight, int64_t txEmbargoStart, int64_t txEmbargoEnd)
+{
+    if (current < txEmbargoStart) {
+        return "pending";
+    } else if (current >= txEmbargoStart && current <= changeHeight) {
+        return "in progress";
+    } else if (current > txEmbargoStart && current >= changeHeight && current <= txEmbargoEnd) {
+        return "in progress";
+    } else if (current > changeHeight && current > txEmbargoEnd) {
+        return "completed";
+    }
+}
+
+std::vector<CUpgrade> getUpgrades()
+{
+    std::vector<CUpgrade> upgrades;
+
+    std::string upgradeid = "hex branchid";
+    std::string name = "Overwinter";
+
+    int64_t current;
+    current = chainActive.Height();
+    int64_t ruleChangeHeight = 380000;
+    int64_t txEmbargoStartHeight = ruleChangeHeight;
+    int64_t txEmbargoEndHeight = ruleChangeHeight + 100;
+
+    std::string status = getUpgradeStatus(current, ruleChangeHeight, txEmbargoStartHeight, txEmbargoEndHeight);
+
+    std::string info = "NU0 / Overwinter: see https://z.cash/overwinter";
+
+    CUpgrade nu0(upgradeid, name, status, ruleChangeHeight, txEmbargoStartHeight, txEmbargoEndHeight, info);
+
+    upgrades.push_back(nu0);
+
+    return upgrades;
+}
```

Comment:
This can alll be removed, and the APIs in `src/consensus/upgrades.h` can be used instead.